### PR TITLE
Bugfix: Allow implicits on non-brands

### PR DESCRIPTION
**Pull request by david-smith-ft**
Brands that have a non-brand source should still provide implicit annotations.
Also, only allow annotations on Content, not on Thing.

### CODE REVIEW

**Comment by david-smith-ft** [comment]
[Allow all brands to show implicits, even if the annotated source is not a brand.](https://trello.com/c/J5u8yG6m/107-allow-all-brands-to-show-implicits-even-if-the-annotated-source-is-not-a-brand)

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,33 @@
-FROM golang:1.8-alpine
+FROM golang:1.10-alpine
```
Comment (tamas-molnar) [inline Dockerfile]: Consider naming the stage, so the copy from commands are more transparent.

File: Dockerfile [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
-  && echo "Fetching dependencies..." \
-  && go get -u github.com/kardianos/govendor \
-  && $GOPATH/bin/govendor sync \
-  && go build -ldflags="${LDFLAGS}" \
-  && mv ${PROJECT} /${PROJECT} \
-  && apk del .build-dependencies \
-  && rm -rf $GOPATH /var/cache/apk/*
+# Set up our extra bits in the image
+RUN apk --no-cache add git curl
+RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (peteclark-ft) [inline Dockerfile]: I don't think we can do this - unless we explicitly run builds with no-cache in Jenkins. Updates to `dep` or alpine packages won't be pulled in if these images are cached.

File: Dockerfile [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
+# Install dependancies and build app
+RUN $GOPATH/bin/dep ensure -vendor-only
+RUN BUILDINFO_PACKAGE="${ORG_PATH}/${PROJECT}/vendor/${ORG_PATH}/service-status-go/buildinfo." \
+    && VERSION="version=$(git describe --tag --always 2> /dev/null)" \
+    && DATETIME="dateTime=$(date -u +%Y%m%d%H%M%S)" \
+    && REPOSITORY="repository=$(git config --get remote.origin.url)" \
+    && REVISION="revision=$(git rev-parse HEAD)" \
+    && BUILDER="builder=$(go version)" \
+    && LDFLAGS="-s -w -X '"${BUILDINFO_PACKAGE}$VERSION"' -X '"${BUILDINFO_PACKAGE}$DATETIME"' -X '"${BUILDINFO_PACKAGE}$REPOSITORY"' -X '"${BUILDINFO_PACKAGE}$REVISION"' -X '"${BUILDINFO_PACKAGE}$BUILDER"'" \
+    && CGO_ENABLED=0 go build -a -installsuffix cgo -o /artifacts/${PROJECT} -ldflags="${LDFLAGS}"
```
Comment (peteclark-ft) [inline Dockerfile]: Why use an `installsuffix` of `cgo`, but set `CGO_ENABLED` to 0 (disabling it)?

File: Dockerfile (hunk 1 above)
Comment (david-smith-ft) [reply Dockerfile]: I did initially, but the Jenkins box doesn't like building dockerfiles with named stages.  https://issues.jenkins-ci.org/browse/JENKINS-44609

File: Dockerfile [hunk 4]
```diff
@@ -1,31 +1,33 @@
...
+
+
+# Multi-stage build - copy only the certs and the binary into the image
+FROM scratch
```
Comment (peteclark-ft) [inline Dockerfile]: Is it ok to use `scratch` with an alpine linux built binary?

Comment (peteclark-ft) [reply Dockerfile]: Otherwise, +1 for multi-stage build use

File: Dockerfile (hunk 1 above)
Comment (tamas-molnar) [reply Dockerfile]: :(

File: Dockerfile (hunk 3 above)
Comment (david-smith-ft) [reply Dockerfile]: I'm going to be honest, George sent me this line and I copied it into the file :P   The installsuffix seems to be a common thing that's been cargo-culted from [here](https://github.com/golang/go/issues/9344).  I'll remove it.

File: Dockerfile (hunk 4 above)
Comment (david-smith-ft) [reply Dockerfile]: Because I staticly build the image, I don't alpine.  Scratch works just fine (as long as you've got the certs).

**Pull request merged**